controllers/host/host_controller.go

```diff
@@ -1543,6 +1544,76 @@ func (r *HostReconciler) ReconcileDeletedHost(client *gophercloud.ServiceClient,
 	return nil
 }
 
+// TriggerDisabledHostReconcilerAndUnlock is used to trigger reconciliation of disabled host.
+// After successful reconciliation unlock-strategy is applied.
+func (r *HostReconciler) TriggerDisabledHostReconcilerAndUnlock(client *gophercloud.ServiceClient, instance *starlingxv1.Host) (err error) {
+	var host *hosts.Host
+	id := instance.Status.ID
+	if id != nil && *id != "" {
+		// This host was previously provisioned so check that it still exists
+		// as the same uuid value; otherwise it may have been deleted and
+		// re-added so we will need to deal with that scenario.
+		host, err = hosts.Get(client, *id).Extract()
+		if err != nil {
+			if _, ok := err.(gophercloud.ErrDefault404); !ok {
+				err = perrors.Wrapf(err, "failed to get: %s", *id)
+				return err
+			}
+
+			// The resource may have been deleted by the system or operator
+			// therefore continue and attempt to recreate it.
+			logHost.Info("resource no longer exists", "id", *id)
```

Comment:
Add a v2 in the final commit if the resource no longer exists is expected

Reply:
Updated accordingly.